packages/react/src/components/ToggleGroup/ToggleGroup.tsx

```diff
@@ -0,0 +1,79 @@
+import type { HTMLAttributes } from 'react';
+import React, { createContext, forwardRef, useId, useState } from 'react';
+import cn from 'classnames';
+
+import { RovingTabindexRoot } from '../../utility-components/RovingTabIndex';
+
+import classes from './ToggleGroup.module.css';
+
+export type ToggleGroupContextProps = {
+  value?: string;
+  defaultValue?: string;
+  onChange?: (value: string) => void;
+  name?: string;
+  size?: 'small' | 'medium' | 'large';
+};
+
+export const ToggleGroupContext = createContext<ToggleGroupContextProps>({});
+
+export type ToggleGroupProps = {
+  /** Value of the ToggleGroup component. Set when you want to controll the value of the component. */
+  value?: string;
+  /** Default value of the ToggleGroup component. Set when you want to use the component as an uncontrolled component. */
+  defaultValue?: string;
+  /** Function that is called when the value of the ToggleGroup component changes. The current value is available in the callback. */
+  onChange?: (value: string) => void;
+  /** Name of the ToggleGroup component. */
+  name?: string;
+  /** Size of the ToggleGroup component. */
+  size?: 'small' | 'medium' | 'large';
+} & Omit<HTMLAttributes<HTMLDivElement>, 'value' | 'onChange'>;
+
```

Comment:
Suggest adding a small jsdoc for the actual component. Look at `Textfield` for example